### DISCUSSION TITLE
Adding typescript to a project that wasn't created with create-next-app doesn't work

### DISCUSSION THREAD

**Discussion by Wassap124** [vercel/next.js#34932] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:41 PST 2021; root:xnu-8019.61.5~1/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.14.0
      npm: 8.3.1
      Yarn: 1.22.17
      pnpm: 6.28.0
    Relevant packages:
      next: 12.1.1-canary.4
      react: 17.0.2
      react-dom: 17.0.2
```

### How are you deploying your application? (if relevant)

next start

### Describe the Bug

Trying to add typescript to an existing next project.  
as specified [in the docs](https://nextjs.org/docs/basic-features/typescript#existing-projects), I ran
```
touch tsconfig.json
```

and then 
```
npm run dev
```

but got no prompt regarding the `typescript` installation and my `tsconfig.json` is still empty

### Expected Behavior

have typescript set up correctly

### To Reproduce

```
npm init
npm i react react-dom next
mkdir -p src/pages
```

what should I install or configure so next could setup typescript automatically for me?

**Comment by mandar1jn** [comment]
just rename your .js files to .ts and your .jsx files to .tsx. Next will do the rest. And add typescript to your dependencies

**Comment by Wassap124** [comment, marked as answer]
that doesn't work...  
When I try to build the project - it doesn't find any of the file names I converted to `.tsx` (because he's looking for their `.jsx` counterpart),  
and running the project without being able to build it yields
``` 
... Try building your app ...
```

**Reply by icyJoseph** [reply]
Hi,

All you have to do next is:

1. Add scripts:

```
 "build": "next build",
 "dev": "next dev",
 "start": "next start",
```

2. Touch `tsconfig.json`
3. Install types for React, Node, and install TypeScript itself 

```shell
npm install --save-dev typescript @types/react @types/node
```

4. Make sure to have something like `src/pages/index.tsx`

```tsx
export const Home = () => <div>Home</div>;

export default Home;
```
5. `npm run dev` - and you ought to be done.

```shell
➜ npm run dev

> test-ts-upgrade@1.0.0 dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
We detected TypeScript in your project and created a tsconfig.json file for you.

event - compiled client and server successfully in 2s (110 modules)
```

**Reply by Wassap124** [reply]
you helped me realize im an idiot, thanks 😂 
used
`next start` instead of `next dev`

**Reply by icyJoseph** [reply]
Please do mark it as answered so it helps people who might run into similar problems.
